File: DependencyInjection/DependencyInjection/_1_ControlFreak/Interfaces/AcmeMessage.cs
using DependencyInjectionBefore._1_ControlFreak.Services;

namespace DependencyInjectionBefore._1_ControlFreak.Interfaces
{
  interface AcmeMessage
  {
    void AuthorizeUsing(ActiveDirectoryBasedAuthorization authorizationRules);
    void WriteTo(DataDestination dataDestination);
  }
}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Interfaces/AcmeMessage.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Services.ActiveDirectoryBasedAuthorization;

public interface AcmeMessage {
  void authorizeUsing(ActiveDirectoryBasedAuthorization authorizationRules);

  void writeTo(DataDestination dataDestination);
}
